subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/inception/BuildSrcChangePerformanceTest.groovy

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.performance.regression.inception
+
+
+import org.gradle.performance.AbstractCrossVersionPerformanceTest
+import org.gradle.performance.categories.SlowPerformanceRegressionTest
+import org.gradle.performance.fixture.JavaTestProject
+import org.gradle.util.GradleVersion
+import org.junit.experimental.categories.Category
+
+@Category(SlowPerformanceRegressionTest)
+class BuildSrcChangePerformanceTest extends AbstractCrossVersionPerformanceTest {
+
+    def setup() {
+        def targetVersion = "6.8-20200927220040+0000"
+        runner.targetVersions = [targetVersion]
+        runner.minimumBaseVersion = GradleVersion.version(targetVersion).baseVersion.version
+    }
+
+    def "buildSrc non-abi change"() {
+        given:
+        runner.tasksToRun = ['help']
+
+        and:
+        JavaTestProject testProject = JavaTestProject.projectFor(runner.testProject)
+        testProject.fileToChange = 'buildSrc/src/main/java/Thing.java'
```

Comment:
I don't think it makes much sense to add the `fileToChange` mutable property to `TestProject`. I think it would be better to add a parameter to `configureForNonAbiChange` passing in the file you want to change. WDYT?

Reply:
You are right. I'll adjust the implementation.